rcl/test/rcl/test_events.cpp

```diff
@@ -206,11 +247,88 @@ wait_for_msgs_and_events(
+      liveliness_policy)) << "Initialized publisher lifespan when unsupported";
+
+    lifespan = {0, 1};
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupSubscriber(deadline, lifespan, lease_duration,
+      liveliness_policy)) << "Initialized subscriber lifespan when unsupported";
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupPublisher(deadline, lifespan, lease_duration,
+      liveliness_policy)) << "Initialized publisher lifespan when unsupported";
+  }
+}
+
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_unsupported_liveliness) {
+  if (is_unsupported) {
+    rmw_time_t deadline {0, 0};
+    rmw_time_t lifespan {0, 0};
+    rmw_time_t lease_duration {0, 0};
+    rmw_qos_liveliness_policy_t liveliness_policy = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE;
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupSubscriber(deadline, lifespan, lease_duration,
+      liveliness_policy)) <<
+      "Initialized subscriber RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE when unsupported";
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupPublisher(deadline, lifespan, lease_duration,
+      liveliness_policy)) <<
+      "Initialized publisher RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_NODE when unsupported";
+
+    liveliness_policy = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC;
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupSubscriber(deadline, lifespan, lease_duration,
+      liveliness_policy)) <<
+      "Initialized subscriber RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC when unsupported";
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupPublisher(deadline, lifespan, lease_duration,
+      liveliness_policy)) <<
+      "Initialized publisher RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC when unsupported";
+  }
+}
+
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_unsupported_unsupported_deadline) {
+  if (is_unsupported) {
+    rmw_time_t deadline;
+    rmw_time_t lifespan {0, 0};
+    rmw_time_t lease_duration {0, 0};
+    rmw_qos_liveliness_policy_t liveliness_policy = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC;
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupSubscriber(deadline, lifespan, lease_duration,
+      liveliness_policy)) << "Initialized subscriber deadline when unsupported";
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupPublisher(deadline, lifespan, lease_duration,
+      liveliness_policy)) << "Initialized publisher deadline when unsupported";
+
+    deadline = {0, 1};
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupSubscriber(deadline, lifespan, lease_duration,
+      liveliness_policy)) << "Initialized subscriber deadline when unsupported";
+    EXPECT_EQ(RMW_RET_ERROR,
+      SetupPublisher(deadline, lifespan, lease_duration,
+      liveliness_policy)) << "Initialized publisher deadline when unsupported";
+  }
+}
+
 /*
  * Basic test of publisher and subscriber liveliness events
  */
 TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_liveliness)
 {
+  if (is_unsupported) {
+    return;
+  }
+  // initialize publisher and subscriber
```

Comment:
I think the function name is obvious enough - this feels redundant. If it feels like you need it then maybe the function should be called `InitializePublisherAndSubscriber`

Reply:
Sounds good.